.github/scripts/env.ps1

```diff
@@ -1,9 +1,12 @@
 # This script enables Developer Command Prompt
 # See https://github.com/microsoft/vswhere/wiki/Start-Developer-Command-Prompt#using-powershell
-$installationPath = vswhere.exe -latest -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath
-if ($installationPath -and (Test-Path "$installationPath\Common7\Tools\vsdevcmd.bat")) {
-  & "${env:COMSPEC}" /s /c "`"$installationPath\Common7\Tools\vsdevcmd.bat`" -arch=x64 -no_logo && set" | ForEach-Object {
-    $name, $value = $_ -split '=', 2
-    "$name=$value" >> $env:GITHUB_ENV
-  }
+$arch = if ($env:BUILD_ARCH -eq "x86_64") { "x64" } else { "arm64" } 
+$installationPath = vswhere.exe -latest -requiresAny -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -requires Microsoft.VisualStudio.Component.VC.Tools.arm64 -property installationPath
```

Comment:
just a guess, maybe `-requiresAny` is causing problems. try something more explicit (didn't test this code, just showing the idea): 

(also, I suggest checking for "arm64" and fallback to "x64", not vice versa.

```
if ($env:BUILD_ARCH -eq "arm64") {
  $arch = "arm64"
  $installationPath = vswhere.exe -latest -requires Microsoft.VisualStudio.Component.VC.Tools.arm64 -property installationPath
} else {
  $arch = "x64"
  $installationPath = vswhere.exe -latest -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath
}
```